service/models.py

```diff
@@ -86,18 +85,20 @@ def delete(cls, supplier_id):
         """ Delete a supplier by it's ID """
         cls.logger.info('Processing deleting for id %s', supplier_id)
         try:
-            return cls.objects(id=supplier_id).delete()
-        except DoesNotExist:
+            res = cls.objects(id=supplier_id).first()
+        except ValidationError:
             return None
+        res.delete()
 
     @classmethod
     def find_by_name(cls, supplier_name):
         """ Find a supplier by its name """
         cls.logger.info('Processing looking for name %s', supplier_name)
         try:
-            return cls.objects.get(supplierName=supplier_name)
-        except DoesNotExist:
+            res = cls.objects.get(supplierName=supplier_name)
+        except ValidationError:
             return None
```

Comment:
For the following testing, let's raise a DataValidationError instead of return None:
```
except ValidationError as e:
            raise DataValidationError('Invalid supplier_name: failed to '
                'find a supplier with given supplier_name')
```

Reply:
Thanks, already done!